build/assets/master/0400_master_daemonset.yaml

```diff
@@ -30,6 +30,7 @@ spec:
           image: $(NODE_FEATURE_DISCOVERY_IMAGE)
           name: nfd-master
           command: ["nfd-master", "--port=12000"]
+          args: ["--port=12000"]
```

Comment:
I think you could/should drop this from here and add it unconditionally in the controller.